### ISSUE TITLE
XGBoost expecting different number of features than custom model

### ISSUE THREAD

**Issue by hspekt** [cox-labs/CoxLab_Bug_Reporting#137]
**Describe the bug**
Using a custom model trained by MQ > Tools > MS/MS intensity prediction errors on library prediction in MaxDIA. 

The error message seems to indicate XGBoost is looking for 227 features but only finding 216.  

**To Reproduce**
Steps to reproduce the behavior using **MQ 2.6.6.0**:

-Create a model using Tools > MS/MS intensity prediction (I used the msms_short.txt provided in a previously closed issue) 
-Specify this custom model to be used by MaxDIA > Library type > Predicted > Custom
-Error on Library_prediction_0

**Error**
 ... Check failed: learner_model_param_.num_feature >= p_fmat->Info().num_col_ (216 vs. 227) : Number of columns does not match number of features in booster._ ...

Many thanks for any ideas or help with this! Are there parameters I can specify for building with the expected 227 features?

**Comment by comaecliptic** [comment]
Hi! Can you attach mqpar.xml file of this run?

**Comment by hspekt** [comment]
[mqpar.xml.zip](https://github.com/user-attachments/files/18184908/mqpar.xml.zip)

Cheers thanks!

**Comment by comaecliptic** [comment]
Yes, specifying a custom model folder only works if the models lie inside the same folder structure as the default ones (so you need to have `standard/hcd/unmodified` inside the folder you set in MQ).
Okay, I don't see anything wrong in mqpar. Did you change any parameters when training the models?

**Comment by hspekt** [comment]
No change to the parameters when training the models, although I did try with and without the default variable modifications.

When I supply the `standard/hcd/unmodified` file structure I get a different error: 

> L:\HS\xxx\combined\proc Library_prediction_0 0 Library_prediction_0 (01/16)  Process 77 0 L:\HS\xxx\mqpar.xml 0 0 16_Never get here._   at MsLib.Fragment.FragmentationTypes.FromName(String fragName)

**Comment by comaecliptic** [comment]
Can you try specifying `standard` as Custom directory? It is working for me.
I'm running now using the same MQ version and the same msms to generate library but still unable to reproduce the issue.
Did you copy the metadata file as well when copying model files to the default directory?
